code/go/0chain.net/smartcontract/zcnsc/mint.go

```diff
@@ -114,11 +113,43 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
 	// record the global nonce from solidity smart contract
 	gn.WZCNNonceMinted[payload.Nonce] = true
 
+	var (
+		amount currency.Coin
+		n      currency.Coin
+		share  currency.Coin
+	)
+	share, _, err = currency.DistributeCoin(gn.ZCNSConfig.MaxFee, int64(len(payload.Signatures)))
+	if err != nil {
+		err = errors.Wrap(err, fmt.Sprintf("%s, DistributeCoin operation, %s", code, info))
+		return
+	}
+	n, err = currency.Int64ToCoin(int64(len(payload.Signatures)))
+	if err != nil {
+		err = errors.Wrap(err, fmt.Sprintf("%s, convert len signatures to coin, %s", code, info))
+		return
+	}
+	amount, err = currency.MinusCoin(payload.Amount, share*n)
```

Comment:
The `share*n` is the `gn.ZCNSconfig.MaxFee`, we don't need to recalculate it. 

So are we going to send `MaxFee` as service to all the authorizers?, if so we'd better rename the `MaxFee` to `AuthorizerServiceFee` or something like to avoid ambiguity.

Reply:
`share*n` can be not equal to `n.ZCNSconfig.MaxFee`, imagine you can't divide maxfee evenly for all authorizers

Reply:
`share*n` can be different from `z.ZCNSconfig.MaxFee`, please check why there is a `share*n` here.

Reply:
I can kind of understand why we name it as `MaxFee` here, as we may send service fee less than it if the `MaxFee` can not be dividend evenly by the authrorizers or validators. If this is the design, I agree that the calculation is correct.

Reply:
@peterlimg we will leave the remaining for client, you can never divide Fee evenly, because you can send different number of signatures, not every time 2f+, it can be more sometimes, should't be an issue